app/presenters/comment_presenter.rb

```diff
@@ -0,0 +1,46 @@
+class CommentPresenter < ApplicationPresenter
+  presents :comment
+  delegate :text, to: :comment
+
+  def submitters_name
+    comment.submitter.name
+  end
+
+  def submitters_link
+    if h.user_signed_in?
+      h.comment_reply_comment_path(comment)
+    else
+      h.user_path(comment.submitter.name)
+    end
+  end
+
+  def link_prefix
+    if h.user_signed_in?
+      "Reply to "
+    else
+      " - contributed by "
+    end
+  end
+
+  def text_trailer
+    h.link_to (link_prefix + submitters_name), submitters_link
+  end
+
+  def score_present?
+    comment&.votes.present?
```

Comment:
Is there a reason why `present?` was used on this AR association here vs. `exists?` in the `replies?` method below?

Reply:
no, not really - `exists?` is more appropriate. Thanks